### DISCUSSION TITLE
Concerned about Isar's long-term viability

### DISCUSSION THREAD

**Discussion by frank06** [isar/isar#1153] [Help]
We recently incorporated Isar as a part of our local persistence strategy. Our app is fairly complex and distributed to iOS and Android devices.

In the coming weeks, more than 10k paid users will be saving their locally generated data to Isar before syncing to a server.

Any issues persisting that data is unacceptable for us because many of our users work offline in remote sites and generate data they can't recover later.

While we only have had success with Isar so far, I'm slightly concerned by the lack of progress in the codebase.

 - There are [several](https://github.com/isar/isar/issues/1011) [fatal](https://github.com/isar/isar/issues/915) unresolved issues, among many others and no indication of progress from the maintainers
 - The lead dev @simc seems to have lost interest in the project as the last release (and commit) are about 4 months old, while he is active on another open source project (Crimson). To be clear, he is absolutely free to do so and I appreciate all of the work he's shared with the world, but I need to point it out. Back in the Hive days, he also disappeared for a very long time until he finally announced he would be moving on to something new: Isar
 - There seem to be other two maintainers, much less prolific, who haven't been very active either

And as far as I know, there is no company providing Isar support. Can anyone confirm? We would be more than happy to pay for 
dedicated support and bug fixes.

Does anyone have a clearer picture of the roadmap for Isar?

Any experiences of companies relying on Isar for critical data?

**Comment by ibrahimdevs** [comment]
@frank06 I totally agree with your opinion. I was using Hive DB, and I completed all migration to isar. Then I faced a fatal issue, it causes random crashes, and it is still waiting. https://github.com/isar/isar/issues/915 and https://github.com/isar/isar/issues/1053 The idea was smart but in terms of maintenance it is not well thought. It is not easy to contribute this package because it uses Rust, ffi, dart .. My final decision is move back to Hive again, at least it is pure dart, I can fix some issues if I need.

@simc thanks for huge amount of work and provide us Hive and Isar. What do you think about this issues and future of Isar and also Hive?

**Comment by simc** [comment]
We started using Isar at ClickUp as the main driver for our app! Rest assured I'm working on Isar almost on a daily basis. Even if it's not always visible.

I have a fix for the two linked issues it will be released shortly.

**Comment by ibrahimdevs** [comment]
Keep calm, king is back! :) @simc thanks for your response and great news. As a maintainer another package, I know that you give this project your precious time and I appreciate for that.

@frank06 you can sponsor to @simc https://github.com/sponsors/simc , it is not too much but may helps him to stay motivated for his packages and shows our appreciate.

**Comment by frank06** [comment]
@ibrahimdevs we'd rather pay (much more) for actual support

**Comment by dayron9110** [comment]
This issue [https://github.com/isar/isar/issues/1068](https://github.com/isar/isar/issues/1068)  happens constantly, could it be reviewed?
